thavalon-server/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# syntax=docker/dockerfile:experimental
+FROM rust:1.46 as builder
+
+COPY . /app
+WORKDIR /app
+
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
```

Comment:
Because of how Docker's caching implementation works, we have to copy any build results out of the directory that's mounted as a cache